assets/src/edit-story/components/panels/link/index.js

```diff
@@ -167,45 +178,51 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
           value={link.url || ''}
           clear
           aria-label={__('Edit: Element link', 'web-stories')}
+          minLength={MIN_MAX.URL.MIN}
+          maxLength={MIN_MAX.URL.MAX}
         />
       </Row>
 
-      {Boolean(link.url) && (
-        <Row>
-          <ExpandedTextInput
-            placeholder={__('Optional description', 'web-stories')}
-            onChange={(value) =>
-              handleChange({ desc: value }, !value /* submit */)
-            }
-            value={link.desc || ''}
-            aria-label={__('Edit: Link description', 'web-stories')}
-          />
-        </Row>
-      )}
-      {Boolean(link.url) && (
-        <Row spaceBetween={false}>
-          <Media
-            value={link.icon || ''}
-            onChange={handleChangeIcon}
-            title={__('Select as link icon', 'web-stories')}
-            buttonInsertText={__('Select as link icon', 'web-stories')}
-            type={'image'}
-            size={64}
-            loading={fetchingMetadata}
-            circle
-          />
-          <IconInfo>
-            <IconText>{__('Optional brand icon', 'web-stories')}</IconText>
-            {link.icon && (
-              <IconRemoveButton
-                onClick={() => handleChange({ icon: null }, true /* submit */)}
-              >
-                <CloseIcon width={14} height={14} />
-                {__('Remove', 'web-stories')}
-              </IconRemoveButton>
-            )}
-          </IconInfo>
-        </Row>
+      {hasSomeLinkContent && (
+        <>
+          <Row>
+            <ExpandedTextInput
+              placeholder={__('Optional description', 'web-stories')}
+              onChange={(value) =>
+                handleChange({ desc: value }, !value /* submit */)
+              }
+              value={link.desc || ''}
+              aria-label={__('Edit: Link description', 'web-stories')}
+            />
+          </Row>
+
+          <Row spaceBetween={false}>
+            <Media
+              value={link.icon || ''}
+              onChange={handleChangeIcon}
+              title={__('Select as link icon', 'web-stories')}
+              buttonInsertText={__('Select as link icon', 'web-stories')}
+              type={'image'}
+              size={64}
+              loading={fetchingMetadata}
+              circle
+              ariaLabel={link.desc}
```

Comment:
The label here should be something like "Edit link icon" -- currently it pulls the description which is not really useful as the edit button label.

E.g. when I use "test.ee" as the link the aria label for the edit button is now "The domain name test.ee is for sale" -- not really useful accessibility information for a button :)

Reply:
FWIW I addressed this in #2254

Reply:
🤦   overlooked the purpose, it should not depend on desc, PTAL now.